DIPs/DIP1xxx-sw.md

```diff
@@ -0,0 +1,954 @@
+
+### Proposed syntax
+
+To understand the proposed changes in a short recap the interesting parts of
+D's grammar will be highlighted.
+Currently [static struct initialization](http://dlang.org/spec/grammar.html#StuctInitializer)
+is defined as follows:
+
+```
+Initializer:
+    VoidInitializer
+    NonVoidInitializer
+
+VoidInitializer:
+    void
+
+NonVoidInitializer:
+    ExpInitializer:
+    ArrayInitializer
+    StructInitializer
+
+StructInitializer:
+    { StructMemberInitializersopt }
+
+StructMemberInitializers:
+    StructMemberInitializer
+    StructMemberInitializer ,
+    StructMemberInitializer , StructMemberInitializers
+
+StructMemberInitializer:
+    NonVoidInitializer
+    Identifier : NonVoidInitializer
+```
+
+#### UnaryExpression
+
+A [UnaryExpression][UnaryExpression] in D is defined as follows:
+
+```
+UnaryExpression:
+    & UnaryExpression
+    ++ UnaryExpression
+    -- UnaryExpression
+    * UnaryExpression
+    - UnaryExpression
+    + UnaryExpression
+    ! UnaryExpression
+    ComplementExpression
+    ( Type ) . Identifier
+    ( Type ) . TemplateInstance
+    DeleteExpression
+    CastExpression
+    PowExpression
+```
+
+The important thing to note is that a [UnaryExpression][UnaryExpression]
+can contain a [PrimaryExpression][PrimaryExpression]
+(via [PowExpression][PowExpression] and [PostfixExpression][PostfixExpression]).
+
+[UnaryExpression]: http://dlang.org/spec/grammar.html#UnaryExpression
+[PowExpression]: http://dlang.org/spec/grammar.html#PowExpression
+[PostfixExpression]: http://dlang.org/spec/grammar.html#PostfixExpression
+[PrimaryExpression]: http://dlang.org/spec/grammar.html#PrimaryExpression
+
+#### PrimaryExpression
+
+A [PrimaryExpression][PrimaryExpression] in D is defined as follows:
+
+```
+PrimaryExpression:
+    Identifier
+    . Identifier
+    TemplateInstance
+    . TemplateInstance
+    this
+    super
+    null
+    true
+    false
+    $
+    IntegerLiteral
+    FloatLiteral
+    CharacterLiteral
+    StringLiterals
+    ArrayLiteral
+    AssocArrayLiteral
+    FunctionLiteral
+    AssertExpression
+    MixinExpression
+    ImportExpression
+    NewExpressionWithArgs
+    BasicTypeX . Identifier
+    Typeof
+    TypeidExpression
+    IsExpression
+    ( Expression )
+    TraitsExpression
+    SpecialKeyword
+```
+
+This addition would extend a [PrimaryExpression][PrimaryExpression] with a `StructLiteral`.
+However, there are multiple potential options for the new syntax:
+
+<a name="option1">
+
+### Option 1: curly braces
+
+```
+StructLiteral:
+
+   Identifier ( StructInitializer )
+```
+
+Example:
+
+    auto s = S({a:42, b:-5});
+
+This syntax matches the static struct initialization the closest.
+
+<a name="option2">
+
+### Option 2: square braces
+
+```
+StructLiteral:
+
+   Identifier ( [ StructMemberInitializers ])
+```
+
+Example:
+
+    auto s = S([a:42, b:-5]);
+
+This syntax may potentially be ambiguous with associative arrays.
```

Comment:
I'm pretty sure this is true. Not sure if it's worth having as a proposal at all.